File: connectors/connector-jira/src/test/scala/com/taskadapter/connector/jira/JiraGTaskBuilder.scala
package com.taskadapter.connector.jira

import java.util.Date

import com.taskadapter.connector.definition.TaskId
import com.taskadapter.model._

object JiraGTaskBuilder {
  def withType(value: String): GTask = {
    val task = new GTask
    task.setValue(Summary, "task " + new Date().getTime)
    task.setValue(TaskType, value)
    task
  }
}

class JiraGTaskBuilder(summary: String = "task " + new Date().getTime) {
  val task = new GTask
  // summary is pretty much always required
  task.setValue(Summary, summary)


  def withDescription(): JiraGTaskBuilder = {
    task.setValue(Description, "description " + new Date().getTime)
    this
  }

  def withPriority(value: Integer): JiraGTaskBuilder = {
    task.setValue(Priority, value)
    this
  }

  def withParentId(value: TaskId): JiraGTaskBuilder = {
    task.setParentIdentity(value)
    this
  }

  def withId(value: Long): JiraGTaskBuilder = {
    task.setId(value)
    this
  }

  def withType(value: String): JiraGTaskBuilder = {
    task.setValue(TaskType, value)
    this
  }

  def build(): GTask = {
    task
  }
}

File: connectors/connector-jira/src/test/java/com/taskadapter/connector/jira/JiraGTaskBuilder.java
package com.taskadapter.connector.jira;

import com.taskadapter.connector.definition.TaskId;
import com.taskadapter.connector.testlib.RandomStringGenerator;
import com.taskadapter.model.AllFields;
import com.taskadapter.model.GTask;

import java.util.Date;

public class JiraGTaskBuilder {
    public static GTask taskWithType(String value) {
        var task = new GTask();
        task.setValue(AllFields.summary, "task " + new Date().getTime());
        task.setValue(AllFields.taskType, value);
        return task;
    }

    public static JiraGTaskBuilder builderWithSummary() {
        return new JiraGTaskBuilder(RandomStringGenerator.randomAlphaNumeric(30));
    }

    private GTask task = new GTask();

    public JiraGTaskBuilder(String summary) {
        // summary is pretty much always required
        task.setValue(AllFields.summary, summary);
    }

    public JiraGTaskBuilder withDescription() {
        task.setValue(AllFields.description, "description " + new Date().getTime());
        return this;
    }

    public JiraGTaskBuilder withPriority(Integer value) {
        task.setValue(AllFields.priority, value);
        return this;
    }

    public JiraGTaskBuilder withParentId(TaskId value) {
        task.setParentIdentity(value);
        return this;
    }

    public JiraGTaskBuilder withId(Long value) {
        task.setId(value);
        return this;
    }

    public JiraGTaskBuilder withType(String value) {
        task.setValue(AllFields.taskType, value);
        return this;
    }

    public GTask build() {
        return task;
    }
}
